src/lib/outline.cc

```diff
@@ -284,7 +284,24 @@ void OutlinePrivate::buildHFCache(OutlineItem * i, int level) {
 		buildHFCache(j, level+1);
 	}
 }
-
+	
+QString Outline::numberToRoman(int number) {
+	// storing roman values of digits from 0-9 
+	// when placed at different places
+	QString m[] = {"", "M", "MM", "MMM"};
+	QString c[] = {"", "C", "CC", "CCC", "CD", "D", "DC", "DCC", "DCCC", "CM"};
+	QString x[] = {"", "X", "XX", "XXX", "XL", "L", "LX", "LXX", "LXXX", "XC"};
+	QString i[] = {"", "I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX"};
+
+	// Converting to roman
+	QString thousands = m[number / 1000];
```

Comment:
What happens when the number of pages becomes more than 3000? Better to use [repeated](https://doc.qt.io/archives/qt-4.8/qstring.html#repeated), see this untested code
```cpp
QString thousands = QString("M").repeated(number / 1000)
```